src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarCommercialButtons.tsx

```diff
@@ -77,12 +74,14 @@ export const ArtworkSidebarCommercialButtons: React.FC<ArtworkSidebarCommercialB
       extractNodes(me.partnerOffersConnection)[0]) ||
     null
 
-  // Fall back to a definitely past value because the timer hook doesn't like nulls
-  const partnerOfferTimer = useTimer(partnerOffer?.endAt || THE_PAST)
   const partnerIcon = artwork.partner?.profile?.icon?.url
 
-  const activePartnerOffer =
-    (!partnerOfferTimer.hasEnded && partnerOffer) || null
+  const isPartnerOfferPast = useMemo(() => {
+    if (!partnerOffer?.endAt) return false
+    return new Date(partnerOffer.endAt) < new Date()
+  }, [partnerOffer])
+
+  const activePartnerOffer = (!isPartnerOfferPast && partnerOffer) || null
```

Comment:
I don't understand why we are removing the timer here. The timer is used because the partner offer is queried once but has an end time, at which time the button should no longer respect it when creating the order.

Reply:
Oh, OK — You're right — I think I understand now. I think this just confused me trying to track down what was causing the re-render. Really it's just the nested component.

Reply:
So I restored the timer code but it definitely is problematic: for instance it continues to check once per second even if the date is past.